katello/katello-restore

```diff
@@ -80,10 +119,97 @@ def restore
   puts "Done.\n"
 end
 
+def display_backup_options
+  puts "---- The given directory does not contain the required files or has too many files ----"
+  puts "---- All backup directories contain: #{STANDARD_BACKUP_FILES.join(", ")} ----"
+  if @is_foremnan_proxy_content
+    puts "---- A #{FOREMAN_PROXY_CONTENT.gsub(/-/, " ")} backup directory contains: #{FOREMAN_PROXY_CONTENT_BACKUP_FILES.join(", ")} ----"
+    puts "---- A #{FOREMAN_PROXY_CONTENT.gsub(/-/, " ")} backup directory contains: #{FOREMAN_PROXY_CONTENT_ONLINE_BACKUP_FILES.join(", ")} ----"
+  else
+    puts "---- An online backup directory contains: #{ONLINE_BACKUP_FILES.join(", ")} ----"
+    puts "---- An offline backup directory contains: #{OFFLINE_BACKUP_FILES.join(", ")} ----"
+  end
+  puts "---- *pulp_data.tar is optional ----"
+  puts "---- Please choose a valid backup directory ----"
+  puts @optparse
+  exit(-1)
+end
+
+def hostname_check
+  # make sure that the system hostname is the same as the backup
+  backup_hostname = `tar zxf #{DIR}/config_files.tar.gz etc/httpd/conf/httpd.conf --to-stdout | grep "ServerName" | awk {'print $2'} | tr -d '"'`.chomp
+  hostname = Socket.gethostname.chomp
+  backup_hostname == hostname
+end
+
+def valid_online_backup
+  if @candlepin_dump_exists && @foreman_dump_exists && @mongo_dump_exists
+    return !@mongo_data_exists && !@pgsql_data_exists
+  else
+    false
+  end
+end
+
+def valid_fpc_online_backup
+  if @mongo_dump_exists
+    return !@mongo_data_exists && !@pgsql_data_exists && !@candlepin_dump_exists && !@foreman_dump_exists
+  else
+    false
+  end
+end
+
+def valid_standard_backup
+  if @mongo_data_exists && @pgsql_data_exists
+    !@candlepin_dump_exists && !@foreman_dump_exists && !@mongo_dump_exists
+  else
+    false
+  end
```

Comment:
nitpick: this can be more cleanly done as:

`@mongo_data_exists && @pgsql_data_exists && !(@candlepin_dump_exists || @foreman_dump_exists || @mongo_dump_exists)`

OR

`@mongo_data_exists && @pgsql_data_exists && !@candlepin_dump_exists && !@foreman_dump_exists && !@mongo_dump_exists`